File: leetcode/[718] Maximum Length of Repeated Subarray/revision.js
/*
 * @lc app=leetcode id=718 lang=javascript
 *
 * [718] Maximum Length of Repeated Subarray
 */

// @lc code=start
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findLength = function(nums1, nums2) {
    const dp = nums1.map(_ => []);
    for (let i = 0; i < nums1.length; i++) {
        for (let j = 0; j < nums2.length; j++) {
            dp[i][j] = nums1[i] === nums2[j] ? (dp[i - 1]?.[j - 1] ?? 0) + 1 : 0;
        }
    }
    return dp.flat().reduce((maxLen, len) => Math.max(maxLen, len));
};
// @lc code=end

File: leetcode/[11] Container With Most Water/revision.java
/*
 * @lc app=leetcode id=11 lang=java
 *
 * [11] Container With Most Water
 */

// @lc code=start
class Solution {
    public int maxArea(int[] height) {
        int maxArea = 0;
        for (int l = 0, r = height.length - 1; r > l;) {
            maxArea = Math.max(Math.min(height[l], height[r]) * (r - l), maxArea);
            if (height[l] < height[r])
                l++;
            else
                r--;
        }
        return maxArea;
    }
}
// @lc code=end
